src/main/java/com/synopsys/integration/jenkins/scan/service/ScanCommandsFactory.java

```diff
@@ -42,6 +46,72 @@ public static ScanPipelineCommands createPipelineCommand(Run<?, ?> run, TaskList
     public static Map<String, Object> preparePipelineParametersMap(SecurityScanStep scanStep) {
         Map<String, Object> parametersMap = new HashMap<>();
 
+        setGlobalConfigurationValues(parametersMap);
+
+        if (!Utility.isStringNullOrBlank(scanStep.getScan_type())) {
+            parametersMap.put(ApplicationConstants.SCAN_TYPE_KEY, scanStep.getScan_type());
+        }
+
+        ScanType scanType = ScanStrategyFactory.getScanType(scanStep.getScan_type());
+        if (scanType.equals(ScanType.COVERITY)) {
+            prepareCoverityParametersMap(scanStep, parametersMap);
+        } else if (scanType.equals(ScanType.POLARIS)) {
+            // preparePolarisParametersMap
+        } else {
+            prepareBlackDuckParametersMap(scanStep, parametersMap);
+        }
+
+        if (!Utility.isStringNullOrBlank(scanStep.getBitbucket_token())) {
+            parametersMap.put(ApplicationConstants.BITBUCKET_TOKEN_KEY, scanStep.getBitbucket_token());
+        }
+
+        prepareBridgeParametersMap(scanStep, parametersMap);
+
+        return parametersMap;
+    }
+
+    private static void setGlobalConfigurationValues(Map<String, Object> parametersMap) {
```

Comment:
Better approach might be returning a new map, rather then mutating the method parameter.

Reply:
Updated accordingly